lightning/src/ln/channel.rs

```diff
@@ -912,7 +923,7 @@ impl<Signer: Sign> Channel<Signer> {
 		if holder_selected_contest_delay < BREAKDOWN_TIMEOUT {
 			return Err(APIError::APIMisuseError {err: format!("Configured with an unreasonable our_to_self_delay ({}) putting user funds at risks", holder_selected_contest_delay)});
 		}
-		let holder_selected_channel_reserve_satoshis = Channel::<Signer>::get_holder_selected_channel_reserve_satoshis(channel_value_satoshis);
+		let holder_selected_channel_reserve_satoshis = Channel::<Signer>::get_holder_selected_channel_reserve_satoshis(channel_value_satoshis, config);
 		if holder_selected_channel_reserve_satoshis < MIN_CHAN_DUST_LIMIT_SATOSHIS {
```

Comment:
This and the other similar `MIN_CHAN_DUST_LIMIT_SATOSHIS` check is now unreachable as the `holder_selected_channel_reserve_satoshis` is lower bound to 1000 now. I therefore suggest removing the two checks.

Reply:
i think those two are separate things and different safety checks. 
MIN_THEIR_CHAN_RESERVE_SATOSHIS is just a safe implementation lower bound, can be changed later. 
Check for MIN_CHAN_DUST_LIMIT_SATOSHIS is protocol limit, and must have a check for it.